extensions/serializer/src/main/java/org/eclipse/dataspaceconnector/serializer/jsonld/JsonldSerializer.java

```diff
@@ -0,0 +1,144 @@
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.serializer.jsonld;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.eclipse.dataspaceconnector.serializer.Serializer;
+import org.eclipse.dataspaceconnector.serializer.jsonld.calendar.XmlGregorianCalendarModule;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+
+/**
+ * Service for de-/serializing java objects to/from JSON-LD string using a customized {@link ObjectMapper}.
+ */
+public class JsonldSerializer implements Serializer<Object, String> {
+
+    private final Monitor monitor;
+    private String context;
+    private Class<?>[] subtypes;
+
+    /**
+     * Constructor.
+     *
+     * @param monitor monitoring service.
+     */
+    public JsonldSerializer(Monitor monitor) {
+        this.monitor = monitor;
+    }
+
+    @Override
+    public String serialize(Object object) throws IOException {
+        return serialize(object, initialize());
+    }
+
+    @Override
+    public Object deserialize(String input, Class<?> type) throws IOException {
+        return deserialize(input, type, initialize());
+    }
+
+    /**
+     * Deserialize string with custom object mapper.
+     *
+     * @param input The string to deserialize.
+     * @param type The class type.
+     * @param objectMapper The custom object mapper.
+     * @return The result object.
+     * @throws IOException if deserialization fails.
+     */
+    public Object deserialize(String input, Class<?> type, ObjectMapper objectMapper) throws IOException {
+        return objectMapper.readValue(input, type);
+    }
+
+    /**
+     * Serialize object with custom object mapper.
+     *
+     * @param object The object to serialize.
+     * @param objectMapper The custom object mapper.
+     * @return The result string.
+     * @throws IOException if serialization fails.
+     */
+    public String serialize(Object object, ObjectMapper objectMapper) throws IOException {
+        String json;
+        try {
+            json = objectMapper.writeValueAsString(object);
+        } catch (JsonProcessingException e) {
+            monitor.warning("Not able to serialize object to JSON-LD");
+            throw e;
+        }
+
+        if (context != null && !context.isEmpty() && !context.isBlank()) {
+            json = addContextToString(json, context);
```

Comment:
I think this could be done without having to manipulate string having the object converted to an `ObjectNode`:
```
var objectNode = objectMapper.convertValue(object, ObjectNode.class);
```

then using the `set` method:
```
objectNode.set("@context", context);
```

And at the end serializing the `ObjectNode`:
```
objectMapper.writeValueAsString(objectNode);
```